packages/bot-web-ui/src/components/trade-animation/__test__/contract-stage-text.spec.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { render, screen } from '@testing-library/react';
+import { contract_stages } from 'Constants/contract-stage';
+import ContractStageText from '../contract-stage-text';
+
+describe('ContractStageText', () => {
```

Comment:
const stageTextMapping = {
        NOT_RUNNING: 'Bot is not running',
        STARTING: 'Bot is starting',
        PURCHASE_SENT: 'Buying contract',
        PURCHASE_RECEIVED: 'Contract bought',
        IS_STOPPING: 'Bot is stopping',
        CONTRACT_CLOSED: 'Contract closed',
    };
    
    Object.keys(stageTextMapping).forEach(stage => {
        it(`should render <ContractStageText /> correct text for ${stage} stage`, () => {
            render(<ContractStageText contract_stage={contract_stages[stage as keyof typeof contract_stages]} />);
            expect(screen.getByText(stageTextMapping[stage])).toBeInTheDocument();
        });
    });
    
    @sandeep-deriv can we do this ?

Reply:
Nice suggestion @rupato-deriv 

It looks more clean and readable with this approach, thanks 🙇